pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -359,18 +359,22 @@ func syncNamespace(
 	kubeClient clientset.Interface,
 	clientPool dynamic.ClientPool,
 	opCache operationNotSupportedCache,
-	groupVersionResources []unversioned.GroupVersionResource,
+	groupVersionResourcesFn func() ([]unversioned.GroupVersionResource, error),
 	namespace *api.Namespace,
 	finalizerToken api.FinalizerName,
 ) error {
+	groupVersionResources, err := groupVersionResourcesFn()
```

Comment:
@liggitt there are multiple calls to perform discovery, so I share your same concerns here.

at minimum, this should only be invoked right before `deleteAllContent`